src/time.c

```diff
@@ -46,6 +46,44 @@ rcutils_time_point_value_as_nanoseconds_string(
   return RCUTILS_RET_OK;
 }
 
+rcutils_ret_t
+rcutils_time_point_value_as_date_string(
+  const rcutils_time_point_value_t * time_point,
+  char * str,
+  size_t str_size)
+{
+  RCUTILS_CHECK_ARGUMENT_FOR_NULL(time_point, RCUTILS_RET_INVALID_ARGUMENT);
+  RCUTILS_CHECK_ARGUMENT_FOR_NULL(str, RCUTILS_RET_INVALID_ARGUMENT);
+  if (0 == str_size) {
+    return RCUTILS_RET_OK;
+  }
+  // best to abs it to avoid issues with negative values in C89, see:
+  //   https://stackoverflow.com/a/3604984/671658
+  uint64_t abs_time_point = (uint64_t)llabs(*time_point);
+  // break into two parts to avoid floating point error
+  uint64_t seconds = abs_time_point / (1000u * 1000u * 1000u);
+  uint64_t nanoseconds = abs_time_point % (1000u * 1000u * 1000u);
+  char str2[21];   // Make sure the buffer is large enough to hold the largest possible uint64_t
+
+  if (rcutils_snprintf(str2, sizeof(str2), "%" PRIu64, nanoseconds)) {
+    RCUTILS_SET_ERROR_MSG("failed to format time point into string as iso8601_date");
+    return RCUTILS_RET_ERROR;
+  }
+
+  time_t now_t = (time_t)(seconds);
+  struct tm ptm = {.tm_year = 2022, .tm_mday = 1};
+  localtime_r(&now_t, &ptm);
```

Comment:
We should check the return value of `localtime_r`, and return an error if it fails.

Further, I'm not 100% sure but I believe this is going to be a problem on Windows.  I think Windows doesn't have `localtime_r`, so there we are probably going to have to use `localtime` or `localtime_s`.  My suggestion is to do the following for now:

```suggestion
#ifdef _WIN32
    localtime_s(&ptm, &now_t);
#else
    localtime_r(&now_t, &ptm);
#endif
```

(though remember to add in error checking as well)

Reply:
I can add the windows change, but [localtime_r](https://en.cppreference.com/w/c/chrono/localtime) doesn't return an error code to check or am I misreading the docs?

Reply:
> I can add the windows change, but [localtime_r](https://en.cppreference.com/w/c/chrono/localtime) doesn't return an error code to check or am I misreading the docs?

It returns an error; see:

> 2-3) copy of the buf pointer, or null pointer on error (which may be a runtime constraint violation or a failure to convert the specified time to local calendar time)

So we need to check for a NULL return.